typescript/vrotsc/e2e/cases/canvas-items/end-item-exception.wf.ts

```diff
@@ -0,0 +1,34 @@
+import { Workflow, In, Out, RootItem, WorkflowEndItem } from "vrotsc-annotations";
+
+@Workflow({
+	name: "Workflow End Exception",
+	path: "VMware/PSCoE",
+	description: "Workflow with root and end item with end mode 1",
+	attributes: {
+		errorMessage: {
+			type: "string"
+		},
+		businessStatus: {
+			type: "string"
+		},
+		endMode: {
+			type: "number"
+		}
+	}
+})
+export class WorkflowEnd {
+
+	@RootItem()
+	public initiateWorkflow() {
+		// NOOP
+	}
+
+	@WorkflowEndItem({
+		endMode: 1,
+		exception: "errorMessage",
+		businessStatus: "Bad"
+	})
+	public workflowEnd(@In endMode: number, @Out errorMessage: string) {
```

Comment:
Why is endMode needed?

Reply:
It is required by the XML representation of the component.

Reply:
This is an input parameter, `end-mode` is a property of the workflow item. It makes no sense to have it as an input too

Reply:
Code updated.